tools/links-check/index.ts

```diff
@@ -9,37 +13,57 @@ main().catch((err: unknown) => {
 });
 
 async function main() {
-  const args = process.argv.slice(2);
+  program
+    .showHelpAfterError()
+    .argument("<paths...>", "Paths of files to be scanned. Supports glob patterns.")
+    .option("--ignore-file <path>", "Path to the ignore file.")
+    .action(async (paths, options) => {
+      let files = await fastGlob(paths);
+      const commonPath = getCommonPath(files);
 
-  if (args.length === 0) {
-    throw new Error("Provide a list of files to scan.");
-  }
+      if (options.ignoreFile) {
+        let ignorePatterns: string[];
 
-  const files = await fastGlob(args);
-  const commonPath = getCommonPath(files);
+        try {
+          const ignoreFileContent = await fs.promises.readFile(path.resolve(options.ignoreFile), "utf-8");
+          ignorePatterns = ignoreFileContent.split("\n").filter(Boolean);
+          console.info(`Using ignore file: ${options.ignoreFile}`);
+        } catch (err) {
+          console.error(`Failed to read ignore file: ${err}`);
+          process.exit(1);
+        }
 
-  console.info();
+        if (ignorePatterns.length) {
+          const ignoreManager = ignore().add(ignorePatterns);
+          files = files.filter((file) => !ignoreManager.ignores(path.relative(commonPath, file)));
```

Comment:
```suggestion
          files = files.filter((file) => !ignoreManager.ignores(file));
```

I think we should check the whole path here, not just the relative part with `commonPath` extracted.
Imagine running from a top-level directory of a repository:
```
cat .gitignore
node_modules/**/

npx @fluffylabs/links-check --ignore-file=.gitignore ./node_modules/**/*.ts
```
My expectation would be that we run on an empty set of files, while I think currently it would run on all `.ts` files, since the `commonPath` would be removed.

Reply:
While writing the comment here I've realised that having `--ignore-file` in a different directory then where we run the command is actually even more tricky. Ignore patters are defined relatively (to the location of the `--ignore-file`), while here we use them as relative to the actual CWD.

I think we should additionally translate the ignore patterns to CWD.

Reply:
I agree that paths being relative to commonPath is not optimal.

We apply the ignore rules relative to the closest common parent directory of the files provided by the user as input which does not equal CWD.

The actual way gitignore works is:
- if the pattern begins with a slash then its treated as relative to the location of `.gitignore`
- if it doesn't begin with a slash, it's applied to the location of `.gitignore` but also any subdirectory

The way i see it, relativizing file paths against the location of the ignore file would mimic `.gitignore` behavior most accurately.

Reply:
I see, thanks for breaking it up.

I think the best way forward would be to:
1. Remove the `commonPath` as suggest in my original comment.
2. Leave the rest as-is.

The reasoning behind this:
1. By requiring the user to specify `--ignore-file` manually, we allow them to "bring in" the ignore file to the CWD they have chosen.
2. So specifying `--ignore-file` is kind of like saying: "I want the patterns from this file I specify to apply to the directory I'm in" - which makes the original location of the `--ignore-file` irrelevant.
3. It makes sense to drop the `commonPath` as well, since the user has full control over the files she's importing as well.
4. Having the ignore file patterns match the actual relative paths the user sees while specifying the command makes the most sense to me.

I'll create another suggestion to update the `--ignore-file` description to reflect that too.

Reply:
done!